pack.go

```diff
@@ -26,53 +26,111 @@ import (
+	// MediaTypeUnknownArtifact is the default artifactType used when no
+	// artifact type is specified.
+	MediaTypeUnknownArtifact = "application/vnd.unknown.artifact.v1"
+)
 
 var (
-	// ErrMissingArtifactType is returned by PackArtifact() when no artifact
-	// type is specified.
-	ErrMissingArtifactType = errors.New("missing artifact type")
-	// ErrInvalidDateTimeFormat is returned by PackArtifact() when
-	// AnnotationArtifactCreated is provided, but its value is not in RFC 3339
-	// format.
-	// Reference: https://datatracker.ietf.org/doc/html/rfc3339#section-5.6
+	// ErrInvalidDateTimeFormat is returned by Pack() when
+	// AnnotationArtifactCreated or AnnotationCreated is provided, but its value
+	// is not in RFC 3339 format.
+	// Reference: https://www.rfc-editor.org/rfc/rfc3339#section-5.6
 	ErrInvalidDateTimeFormat = errors.New("invalid date and time format")
 )
 
 // PackOptions contains parameters for oras.Pack.
 type PackOptions struct {
+	// Subject is the subject of the manifest.
+	Subject *ocispec.Descriptor
+	// ManifestAnnotations is the annotation map of the manifest.
+	ManifestAnnotations map[string]string
+
+	// PackImageManifest controls whether to pack an image manifest or not.
+	//   - If true, pack an image manifest; artifactType will be used as the
+	// the config media type of the image manifest.
+	// media type.
+	//   - If false, pack an artifact manifest.
+	// Default: false.
+	PackImageManifest bool
 	// ConfigDescriptor is a pointer to the descriptor of the config blob.
+	// If specified, artifactType will be implied by the media type of the
+	// specified descriptor, and ConfigAnnotations will be ignored.
+	// This option is valid only when PackImageManifest is true.
 	ConfigDescriptor *ocispec.Descriptor
-	// ConfigMediaType is the media type of the config blob.
-	// If not specified, MediaTypeUnknownConfig will be used.
-	ConfigMediaType string
 	// ConfigAnnotations is the annotation map of the config descriptor.
+	// This option is valid only when PackImageManifest is true,
+	// and ConfigDescriptor is nil.
 	ConfigAnnotations map[string]string
-	// ManifestAnnotations is the annotation map of the manifest.
-	ManifestAnnotations map[string]string
 }
 
-// PackArtifactOptions contains parameters for oras.PackArtifact.
-type PackArtifactOptions struct {
-	// Subject is the subject of the ORAS Artifact Manifest.
-	Subject *artifactspec.Descriptor
-	// ManifestAnnotations is the annotation map of the manifest.
-	ManifestAnnotations map[string]string
+// Pack packs the given blobs, generates a manifest for the pack,
+// and pushes it to a content storage.
+// If succeeded, returns a descriptor of the manifest.
+func Pack(ctx context.Context, pusher content.Pusher, artifactType string, blobs []ocispec.Descriptor, opts PackOptions) (ocispec.Descriptor, error) {
+	if opts.PackImageManifest {
+		return packImage(ctx, pusher, artifactType, blobs, opts)
+	}
+	return packArtifact(ctx, pusher, artifactType, blobs, opts)
```

Comment:
nit: Should we rename the variable `artifactType`? As it is artifact type in `PackImage` and media type in `PackArtifact`. A more generic name like `type` may be less confusing.

Reply:
Yeah this is confusing... But `type` is also confusing😂 (naming is hard!).
Maybe we should add more docs here.